magma/bfloat.py

```diff
@@ -61,11 +63,11 @@ def declare_ite(cls, T):
         if N != 16:
             raise NotImplementedError("Only BFloat16 supported")
         coreir_genargs = {"exp_bits": 8, "frac_bits": 7}
-        return m.circuit.DeclareCoreirCircuit(f"magma_BFloat_{N}_ite_{t_str}",
-                                              "I0", m.In(T),
-                                              "I1", m.In(T),
-                                              "S", m.In(m.Bit),
-                                              "O", m.Out(T),
-                                              coreir_name="mux",
-                                              coreir_genargs=coreir_genargs,
-                                              coreir_lib="float")
+        return DeclareCoreirCircuit(f"magma_BFloat_{N}_ite_{t_str}",
+                                    "I0", In(T),
+                                    "I1", In(T),
+                                    "S", In(Bit),
+                                    "O", Out(T),
+                                    coreir_name="mux",
+                                    coreir_genargs=coreir_genargs,
+                                    coreir_lib="float")
```

Comment:
I think Declare will generate a deprecation warning, so at some point we should update this to the subclass pattern.

Reply:
Definitely; will put up a follow up PR to get rid of more of these.